frontend/src/components/smart-contracts/SmartContractContext.ts

```diff
@@ -0,0 +1,20 @@
+import { createContext } from 'react';
+import {
+  SmartContractState,
+  initialSmartContractsState,
+  SmartContractAction,
+} from 'components/smart-contracts/SmartContractReducer';
+
+export type ISmartContractContext = {
+  contractsState: SmartContractState;
+  dispatch: (arg0: SmartContractAction) => void;
+};
+
+const defaultSmartContractsContextValue = {
+  contractsState: initialSmartContractsState,
+  dispatch: () => {},
```

Comment:
Instead of exporting `dispatch`, could we make methods that dispatch? It will just make the component code a lot easier if we can call `attachWallet()`

Reply:
I removed the reducer entirely.  It now includes a function named `updateERC20Approval`